projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -2998,7 +2998,7 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
      * @internal
      */
     public setFilteredSortedData(data, pinned: boolean) {
-        data = data.map(rec => rec.ghostRecord !== undefined ? rec.recordRef : rec);
+        data = data ? data.map(rec => rec.ghostRecord !== undefined ? rec.recordRef : rec) : [];
```

Comment:
After testing this I noticed that the data is never passed wrapped as a ghost record, hence the mapping here seems unnecessary. The whole line can be removed.